content/docs/rfcs/35/README.md

```diff
@@ -0,0 +1,320 @@
+    - `transport-message-len-len` and `transport-message-len` are set accordingly to `transport-message` length;
+    - `transport-message-ad` contains the authentication data for `ct`.
+
+
+### Public Keys Serialization
+
+Diffie-Hellman public keys can be trasmitted in clear 
+or in encrypted form with authentication data attached. 
+To distinguish between these two cases, public keys are serialized as the concatenation of the following three fields:
+
+- `flag`: 
+is equal to `1` if the public key is encrypted; 
+`0` otherwise (1 byte);
+- `pk`: 
+if `flag = 0`, it contains an encoding of the X coordinate of the public key. 
+If `flag = 1`, it contains a symmetric encryption of an encoding of the X coordinate of the public key;
+- `pk_ad`: 
+if `flag = 0`, it is empty; 
+if `flag = 1`, it contains the authentication data for `pk`;
+
+
+As regards the underlying supported [cryptographic primitives](#Cryptographic-primitives):
+- `Curve25519` public keys X coordinates are encoded in little-endian as 32 bytes arrays;
+- `ChaChaPoly` authentication data consists of 16 bytes 
+(nonces are implicitely defined by Noise [processing rules](http://www.noiseprotocol.org/noise.html#processing-rules)).
+
+In all supported Noise protocols, 
+parties' static keys are transmitted encrypted, 
+while ephemeral keys MAY be encrypted after a handshake is complete.
+
+
+### Padding
+
+To prevent some metadata leakage, 
+encrypted transport messages SHOULD be padded before encryption 
+to a multiple of the underlying symmetric cipher block size 
+(16 bytes for `ChaChaPoly`).
+
+It is therefore recommended to pad transport messages to a multiple of 256 bytes.
+
+
+## After-handshake
+
+We note that 
+during the initial 1-RTT communication, 
+handshake messages can be linked to the respective parties 
+through the `contentTopic` employed for such communication. 
+
+After a handshake is completed, 
+parties SHOULD derive from their shared secret key (preferably using `HKDF`) a new random `contentTopic` 
+and continue their communication there.
+
+When communicating on the new `contentTopic`, 
+parties SHOULD set `protocol-id` to `0` 
+to reduce metadata leakages and indicate that the message is an *after-handshake* message.
+
+It is recommended that each party attaches an (unencrypted) ephemeral key in `handshake-message` to every message sent.
+According to [Noise processing rules](http://www.noiseprotocol.org/noise.html#processing-rules), 
+this allows updates to the shared secret key 
+by hashing the result of an ephemeral-ephemeral Diffie-Hellman exchange every 1-RTT communication.
+
+
+## Backward Support for Symmetric/Asymmetric Encryption
+
+It is possible to have backward compatibility to symmetric/asymmetric encryption primitives from [26/WAKU-PAYLOAD](/spec/26), 
+effectively encapsulating  payload encryption [14/WAKU-MESSAGE version 1](/spec/14/#version1) in [version 2](/spec/14/#version2).
```

Comment:
```suggestion
effectively encapsulating payload encryption [14/WAKU-MESSAGE version 1](/spec/14/#version1) in [version 2](/spec/14/#version2).
```

Reply:
Addressed in https://github.com/vacp2p/rfc/pull/496/commits/b587e8b2cb05ce1fa6042c2417696cb50be6fcf4